### ISSUE TITLE
Hidden diagnostics for failure to inline

### ISSUE THREAD

**Issue by Smattr** [compiler-explorer/compiler-explorer#261]
Consider the following code:

    #include <immintrin.h>

    void foo() {
      __m128i x;
      _mm_testz_si128(x, x);
    }

GCC 6.1.0 says the following. Paths have been changed to protect the innocent.

    In file included from /.../include/immintrin.h:37:0,
                 from foo.c:
    /.../include/smmintrin.h: In function ‘void foo()’:
    /.../include/smmintrin.h:67:1: error: inlining failed in call to always_inline ‘int _mm_testz_si128(__m128i, __m128i)’: target specific option mismatch
    _mm_testz_si128 (__m128i __M, __m128i __V)
    foo.c: note: called from here
             _mm_testz_si128(x, x);

Compiler explorer on the other hand says:

    !! note: called from here

It doesn't seem possible to get at the underlying error through the web interface. I would hazard a guess that this has nothing to do with the intrinsic or this specific category of error, but is related to the format of the diagnostic.

Possibly a duplicate of #189?

**Comment by mattgodbolt** [comment]
Right; this is a problem: as you note I don't parse errors originating from external files. Will have a think; thanks.

**Comment by mattgodbolt** [comment]
I am not 100% sure when this happened, but this _seems_ to be fixed. https://godbolt.org/z/xBqSwC now says:

```
In file included from /opt/compiler-explorer/gcc-6.1.0/lib/gcc/x86_64-linux-gnu/6.1.0/include/immintrin.h:37:0,
                 from <source>:1:
/opt/compiler-explorer/gcc-6.1.0/lib/gcc/x86_64-linux-gnu/6.1.0/include/smmintrin.h: In function 'void foo()':
/opt/compiler-explorer/gcc-6.1.0/lib/gcc/x86_64-linux-gnu/6.1.0/include/smmintrin.h:67:1: error: inlining failed in call to always_inline 'int _mm_testz_si128(__m128i, __m128i)': target specific option
mismatch
 _mm_testz_si128 (__m128i __M, __m128i __V)
 ^~~~~~~~~~~~~~~
<source>:5:24: note: called from here
   _mm_testz_si128(x, x);
                        ^
Compiler returned: 1
```
